src/H5Dvirtual.c

```diff
@@ -3672,3 +3792,254 @@ H5D__virtual_release_source_dset_files(H5D_virtual_held_file_t *head)
 done:
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5D__virtual_release_source_dset_files() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5D__rtree_should_insert
+ *
+ * Purpose:     Determine whether the given mapping is valid for
+ *              insertion into a spatial tree
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5D__rtree_should_insert(void *mapping_entry, bool *should_insert)
```

Comment:
This should either be a void function, a macro, or it should be converted to handle errors normally. I'm pretty sure the checks below can be assertions, or eliminated if this is turned into a macro.